File: individual_jsp/src/java/servlet/Checkin.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package servlet;
import java.math.BigDecimal;
import java.sql.*;
import conns.ConnSql;
import java.io.IOException;
import java.io.PrintWriter;
 
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
 
@WebServlet("/Checkin")
public class Checkin extends HttpServlet {
	private static final long serialVersionUID = 1L;
 
	@Override
	protected void service(HttpServletRequest request,HttpServletResponse response) throws ServletException, IOException{
		response.setContentType("text/html;charset=utf-8");
		PrintWriter out = response.getWriter();
		request.setCharacterEncoding("utf-8");
		String bid = request.getParameter("bid");
		String tid = request.getParameter("tid");
                String sid = request.getParameter("sid");
              
                         try{
                          ConnSql test=new ConnSql();
                          Connection one=test.getConnection();//获取getConnection()方法
                          Statement statement = one.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,ResultSet.CONCUR_UPDATABLE);//获取操作数据库的对象
                         // 
                          String sql="select * from checkin where bid='"+bid+"'and sid='"+sid+"'"; 
                          ResultSet resultSet = statement.executeQuery(sql);//执行sql，获取结果集
                          resultSet.last();
                          int lastRow=resultSet.getRow();
                          if(lastRow>0){
                             out.println("<script>alert('You have registered for this course');window.history.back();</script>");
                          }else{
                          //out.println("<script>alert('2');window.history.back();</script>");
                          java.sql.Timestamp stamp=new java.sql.Timestamp(System.currentTimeMillis());
                           String sqls="INSERT INTO checkin (tid,bid,sid,state,dates) VALUES('"+tid+"','"+bid+"','"+sid+"','2','"+stamp+"')";
                          int resultSets = statement.executeUpdate(sqls);//执行sql，获取结果集
                          if(resultSets>0){
                                  
                                  out.println("<script>alert('Successful registration');window.location.href='web/student/subject.jsp';</script>");
                                 
                          
                          }else{
                          
                           out.println("<script>alert('error');window.history.back();</script>");
                          
                          }
                          }
                          
                          
                          
                         }catch(Exception e){
//                          out.println("<script>alert('error！');window.history.back();</script>");
                          out.println(e.getMessage());
                           
                         }
                             
                           
                         
                          
			
		}
		
	
 
}

File: individual_php/student/checkin.php
<?php
session_start();
require_once ('../db.php');
$user=$_SESSION['student'];
$sid=$_SESSION['sid'];
$start_times=$_GET['start_time'];
$id=$_GET['id'];
$tid=$_GET['tid'];
//比较时间


				$sql = "select * from subject where bid ='$id'"  ;//数据库查询
				mysqli_query($conn,'set names utf8');
			       $res=mysqli_query($conn,$sql);//查询出数据并赋值
			       header("Content-type: text/html; charset=utf-8");//头部文件	
				$ros=mysqli_fetch_assoc($res);
				$nowtime=date('Y-m-d H:m');
				$ctime=date('Y-m-d ').$ros['start_time'];
				$btime=date('Y-m-d ').$ros['end_time'];
				$n= strtotime($nowtime)."<br/>";
				$b= strtotime($btime);
				$c= strtotime($ctime);//开始时间
				echo $c."</br/>";
				echo $b."</br/>";
				echo $n."</br/>";
				
				if($n<$c or $n>$b ){
					
					echo"<script>alert('It is past the check-in time'); window.location.href='subject.php';</script>";
				}else{
					
					$check="select * from checkin where bid ='$id' and sid='$sid'" ;
					$checkres=mysqli_query($conn,$check);//查询出数据并赋值
					$checkros=mysqli_fetch_assoc($checkres);
					if(!isset($checkros['bid'])){
						$dates=date('Y-m-d,H:m:s');
						$updates="INSERT INTO checkin(tid,bid,sid,state,dates)VALUES('$tid', '$id', '$sid', '2','$dates')";//数据库查询
//						mysqli_query($conn,$updates);
//                                                echo mysqli_error(conn);
						if(mysqli_query($conn,$updates)){
						    echo"<script>alert('签到成功！'); window.location.href='subject.php';</script>";
						}else{
						   echo"<script>alert('签到失败！'); window.location.href='subject.php';</script>";
						}

					}else{
						
					echo"<script>alert('You have signed up in the same period'); window.location.href='subject.php';</script>";	
						
					}
							
}
				
	
	
	



?>